scripts/rename_resources.py

```
@@ -0,0 +1,52 @@
+    exit(1)
+
+renamed = {
+    'aws_cloudwatch_log_group.azul_error_log': 'aws_cloudwatch_log_group.error_log',
+    'aws_cloudwatch_log_group.azul_index_log': 'aws_cloudwatch_log_group.index_log',
+    'aws_cloudwatch_log_group.azul_search_log': 'aws_cloudwatch_log_group.search_log',
+    'aws_cloudwatch_log_resource_policy.azul_es_cloudwatch_policy': 'aws_cloudwatch_log_resource_policy.index',
+    'aws_dynamodb_table.cart_items_table': 'aws_dynamodb_table.cart_items',
+    'aws_dynamodb_table.carts_table': 'aws_dynamodb_table.carts',
+    'aws_dynamodb_table.users_table': 'aws_dynamodb_table.users',
+    'aws_dynamodb_table.versions_table': 'aws_dynamodb_table.object_versions',
+    'aws_elasticsearch_domain.elasticsearch[0]': 'aws_elasticsearch_domain.index',
+    'aws_iam_role.state_machine_iam_role': 'aws_iam_role.states',
+    'aws_iam_role_policy.state_machine_iam_policy': 'aws_iam_role_policy.states',
+    'aws_s3_bucket.bucket': 'aws_s3_bucket.storage',
+    'aws_s3_bucket.url_bucket': 'aws_s3_bucket.urls',
+    'aws_sfn_state_machine.cart_export_state_machine': 'aws_sfn_state_machine.cart_export',
+    'aws_sfn_state_machine.cart_item_state_machine': 'aws_sfn_state_machine.cart_item',
+    'aws_sfn_state_machine.manifest_state_machine': 'aws_sfn_state_machine.manifest',
+    'null_resource.hmac-secret': 'null_resource.hmac_secret'
+}
+
+run_args = {
+    'cwd': TERRAFORM_PATH,
+    'check': True,
+    'capture_output': True,
+    'shell': False
+}
+
+if __name__ == '__main__':
+    configure_script_logging(log)
+    tf_state = subprocess.run(['terraform', 'state', 'list'], **run_args)
+    current = [r.strip() for r in tf_state.stdout.decode('utf-8').split('\n')]
+    for resource in filter(lambda r: r in renamed.keys(), current):
+        log.info('Found %r, renaming to %r', resource, renamed[resource])
+        subprocess.run(['terraform', 'state', 'mv', resource, renamed[resource]],
```

What happens if you run this twice? Is `terraform state mv` idempotent?

---

`terraform state mv` is idempotent and will fail if trying to rename a nonexistent resource

---

If it fails, it is NOT idempotent. Do we need to swallow the error or do we want to fail since a check above should prevent the case?

---

Summary from standup: yes, because `terraform state mv` should only be run on resources that need to be renamed. If `terraform state mv` fails, that's indicative of a larger problem (maybe a bug in the aforementioned check) that requires attention.